pkg/kubelet/kubelet.go

```diff
@@ -2591,8 +2625,21 @@ func (kl *Kubelet) HandlePodReconcile(pods []*v1.Pod) {
 func (kl *Kubelet) HandlePodSyncs(pods []*v1.Pod) {
 	start := kl.clock.Now()
 	for _, pod := range pods {
-		mirrorPod, _ := kl.podManager.GetMirrorPodByPod(pod)
-		kl.dispatchWork(pod, kubetypes.SyncPodSync, mirrorPod, start)
+		pod, mirrorPod, wasMirror := kl.podManager.GetPodAndMirrorPod(pod)
+		if wasMirror {
+			if pod == nil {
+				klog.V(2).InfoS("Unable to find pod for mirror pod, skipping", "mirrorPod", klog.KObj(mirrorPod), "mirrorPodUID", mirrorPod.UID)
+				continue
+			}
+			// since we are syncing all pods, we don't have to sync mirrors
```

Comment:
Before, if we got passed a mirror pod, we would send the mirror pod to UpdatePod, which is not correct.  pod workers should never receive a mirror pod in the `Pod` field.  HandlePodSyncs only accepts "real pods".  So this would be programmer error (even if it never happened).  I can add a klog.

Reply:
Got it, it makes sense we shouldn't sync mirror pods to `UpdatePod`. I was confused because I saw we _were_ doing that before, but it sounds like that was a bug and we shouldn't have been doing that in the first place.